lib/src/baseFormValidation.ts

```diff
@@ -1,20 +1,76 @@
-import { } from 'core-js';
-import { IValidationEngine, ValidationEngine } from './validationEngine';
-import { FieldValidationResult, FormValidationResult } from './entities';
+import { ValidationEngine } from './validationEngine';
+import {
+  ValidationConstraints,
+  FormValidationFunction,
+  FieldValidationResult,
+  FormValidationResult,
+} from './entities';
 import { consts } from './consts';
 
-export class BaseFormValidation {
-  _validationEngine: IValidationEngine;
+interface FormValidation {
+  validateField(vm: any, key: string, value: any, filter?: any): Promise<FieldValidationResult>;
+  validateForm(vm: any): Promise<FormValidationResult>;
+  isValidationInProgress(): boolean;
+  isFormDirty(): boolean;
+  isFormPristine(): boolean;
+  addFieldValidation(key: string, validation: (value: string, vm: any) => FieldValidationResult, filter?: any): FormValidation;
+  addFieldValidationAsync(key: string, validation: (value: string, vm: any) => Promise<FieldValidationResult>, filter?: any): FormValidation;
+}
+
+export class BaseFormValidation implements FormValidation {
+  private validationEngine: ValidationEngine;
+  constructor(validationConstraints: ValidationConstraints) {
+    this.validationEngine = new ValidationEngine();
+    this.parseValidationConstraints(validationConstraints);
+  }
+
+  private parseValidationConstraints(validationConstraints: ValidationConstraints) {
+    if (validationConstraints && typeof validationConstraints === 'object') {
+      if (validationConstraints.hasOwnProperty('global') && validationConstraints.global instanceof Array) {
```

Comment:
Why you use `validationConstraints.hasOwnProperty('global')`?

You can't use `validationConstraints.global`?